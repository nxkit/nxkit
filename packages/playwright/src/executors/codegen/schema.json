{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "title": "Codegen executor",
  "description": "",
  "type": "object",
  "properties": {},
  "required": []
}
