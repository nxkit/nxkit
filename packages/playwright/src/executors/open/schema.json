{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "title": "Open executor",
  "description": "",
  "type": "object",
  "properties": {},
  "required": []
}
