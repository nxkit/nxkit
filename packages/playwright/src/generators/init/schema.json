{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "Init",
  "title": "",
  "type": "object",
  "properties": {
    "skipFormat": {
      "type": "boolean",
      "description": "Skip formatting files",
      "default": false
    },
    "skipPackageJson": {
      "type": "boolean",
      "description": "Do not add dependencies to `package.json`",
      "default": false
    },
    "skipPlaywrightInstall": {
      "type": "boolean",
      "description": "Do not run playwright install command",
      "default": false
    },
    "forcePlaywrightInstall": {
      "type": "boolean",
      "description": "Force reinstall of stable browser channels",
      "default": false,
      "alias": "f"
    }
  },
  "required": []
}
