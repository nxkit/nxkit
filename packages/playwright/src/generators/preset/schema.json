{
  "$schema": "http://json-schema.org/schema",
  "$id": "Preset",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use?"
    },
    "baseUrl": {
      "type": "string",
      "description": "The base url of the application to be tested.",
      "default": "http://localhost:4200",
      "x-prompt": "What is the base URL to test?"
    },
    "linter": {
      "description": "The tool to use for running lint checks.",
      "type": "string",
      "enum": ["eslint", "none"],
      "default": "eslint"
    }
  },
  "required": ["name", "baseUrl"]
}
