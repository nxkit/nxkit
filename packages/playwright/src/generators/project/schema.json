{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "Project",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "Name of the e2e project to create",
      "$default": {
        "$source": "argv",
        "index": 0
      }
    },
    "frontendProject": {
      "type": "string",
      "description": "The name of the frontend project to test"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the project (used for linting)",
      "alias": "t"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed",
      "alias": "d",
      "x-completion-type": "directory"
    },
    "skipFormat": {
      "description": "Skip formatting files.",
      "type": "boolean",
      "default": false
    }
  },
  "required": []
}
