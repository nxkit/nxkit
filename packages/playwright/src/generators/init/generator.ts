import {
  formatFiles,
  GeneratorCallback,
  Tree,
  runTasksInSerial,
} from '@nx/devkit';
import { initGenerator as jsInitGenerator } from '@nx/js';
import { addDependencies } from './lib/add-dependencies';
import { installPlaywright } from './lib/install-playwright';
import { updateGitIgnore } from './lib/update-gitignore';
import { InitGeneratorSchema } from './schema';

export async function initGenerator(tree: Tree, options: InitGeneratorSchema) {
  const tasks: GeneratorCallback[] = [];

  if (!options.skipPackageJson) {
    const installPackagesTask = addDependencies(tree);
    tasks.push(installPackagesTask);
  }

  if (!options.skipPlaywrightInstall) {
    const installPlaywrightTask = await installPlaywright({
      force: options.forcePlaywrightInstall,
    });
    tasks.push(installPlaywrightTask);
  }

  await jsInitGenerator(tree, {
    skipFormat: true,
  });

  updateGitIgnore(tree);

  if (!options.skipFormat) {
    await formatFiles(tree);
  }

  return runTasksInSerial(...tasks);
}

export default initGenerator;
