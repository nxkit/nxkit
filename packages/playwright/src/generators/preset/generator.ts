import { names, Tree, runTasksInSerial } from '@nx/devkit';
import projectGenerator from '../project/generator';
import { PresetGeneratorSchema } from './schema';

function normalizeOptions(
  tree: Tree,
  options: PresetGeneratorSchema
): PresetGeneratorSchema {
  const name = names(options.name).fileName;
  return {
    ...options,
    name,
  };
}

export async function presetGenerator(
  tree: Tree,
  options: PresetGeneratorSchema
) {
  const normalizedOptions = normalizeOptions(tree, options);
  const projectTask = await projectGenerator(tree, normalizedOptions);
  return runTasksInSerial(projectTask);
}

export default presetGenerator;
