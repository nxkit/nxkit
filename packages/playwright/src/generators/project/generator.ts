import { formatFiles, Tree } from '@nrwl/devkit';
import { runTasksInSerial } from '@nrwl/workspace/src/utilities/run-tasks-in-serial';
import initGenerator from '../init/generator';
import { addLinter } from './lib/add-linter';
import { addProjectConfig } from './lib/add-project-config';
import { addProjectFiles } from './lib/add-project-files';
import { normalizeOptions } from './lib/normalize-options';
import { ProjectGeneratorSchema } from './schema';

export default async function (tree: Tree, options: ProjectGeneratorSchema) {
  const normalizedOptions = normalizeOptions(tree, options);

  const initTask = await initGenerator(tree, {
    skipFormat: true,
    skipPackageJson: false,
  });

  addProjectConfig(tree, normalizedOptions);
  addProjectFiles(tree, normalizedOptions);

  const lintTask = await addLinter(tree, normalizedOptions);

  if (!normalizedOptions.skipFormat) {
    await formatFiles(tree);
  }

  return runTasksInSerial(initTask, lintTask);
}
