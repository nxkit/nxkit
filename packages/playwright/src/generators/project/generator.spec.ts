import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';
import { Tree, readProjectConfiguration } from '@nrwl/devkit';

import generator from './generator';
import { ProjectGeneratorSchema } from './schema';

describe('project generator', () => {
  let appTree: Tree;
  const options: ProjectGeneratorSchema = { name: 'test' };

  beforeEach(() => {
    appTree = createTreeWithEmptyWorkspace();
  });

  it('should run successfully', async () => {
    await generator(appTree, options);
    const config = readProjectConfiguration(appTree, 'test');
    expect(config).toBeDefined();
  });
});
