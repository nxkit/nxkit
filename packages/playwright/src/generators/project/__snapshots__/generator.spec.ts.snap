// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Playwright Project --directory should set right path names in \`playwright.config.ts\` 1`] = `
"import type { PlaywrightTestConfig } from '@playwright/test';
import type { PlaywrightTestConfig } from '@playwright/test';
import { devices } from '@playwright/test';

/**
 * Read environment variables from file.
 * https://github.com/motdotla/dotenv
 */
// require('dotenv').config();

/**
 * See https://playwright.dev/docs/test-configuration for more information
 */

const baseURL = process.env.E2E_BASE_URL || 'http://localhost:4200/';

export const config: PlaywrightTestConfig = {
  testDir: './src/e2e',
  retries: process.env.CI ? 2 : 0,
  maxFailures: 2,
  timeout: 30 * 1000,
  expect: {
    timeout: 5000
  },
  use: {
    actionTimeout: 0,
    baseURL,
    trace: 'on-first-retry',
  },
  reporter: [
    ['html', { outputFolder: '../../../dist/apps/my-dir/my-app-e2e/playwright-report' }],
    ['json', { outputFile: '../../../dist/apps/my-dir/my-app-e2e/playwright-report/test-results.json' }],
  ],
  fullyParallel: true,
  forbidOnly: !!process.env.CI,
  workers: process.env.CI ? 1 : undefined,
  projects: [
    {
      name: 'chromium',
      use: {
        ...devices['Desktop Chrome'],
      },
    },
    {
      name: 'firefox',
      use: {
        ...devices['Desktop Firefox'],
      },
    },
    {
      name: 'webkit',
      use: {
        ...devices['Desktop Safari'],
      },
    },
    /* Test against mobile viewports. */
    // {
    //   name: 'Mobile Chrome',
    //   use: {
    //     ...devices['Pixel 5'],
    //   },
    // },
    // {
    //   name: 'Mobile Safari',
    //   use: {
    //     ...devices['iPhone 12'],
    //   },
    // },
    /* Test against branded browsers. */
    // {
    //   name: 'Microsoft Edge',
    //   use: {
    //     channel: 'msedge',
    //   },
    // },
    // {
    //   name: 'Google Chrome',
    //   use: {
    //     channel: 'chrome',
    //   },
    // },
  ],
};

export default config;
"
`;

exports[`Playwright Project --directory should set right path names in \`tsconfig.e2e.json\` 1`] = `
Object {
  "compilerOptions": Object {
    "allowJs": true,
    "outDir": "../../../dist/out-tsc",
    "sourceMap": false,
    "types": Array [
      "node",
    ],
  },
  "extends": "./tsconfig.json",
  "include": Array [
    "src/**/*.ts",
    "src/**/*.js",
    "playwright.config.ts",
  ],
}
`;

exports[`Playwright Project should set right path names in \`playwright.config.ts\` 1`] = `
"import type { PlaywrightTestConfig } from '@playwright/test';
import type { PlaywrightTestConfig } from '@playwright/test';
import { devices } from '@playwright/test';

/**
 * Read environment variables from file.
 * https://github.com/motdotla/dotenv
 */
// require('dotenv').config();

/**
 * See https://playwright.dev/docs/test-configuration for more information
 */

const baseURL = process.env.E2E_BASE_URL || 'http://localhost:4200/';

export const config: PlaywrightTestConfig = {
  testDir: './src/e2e',
  retries: process.env.CI ? 2 : 0,
  maxFailures: 2,
  timeout: 30 * 1000,
  expect: {
    timeout: 5000
  },
  use: {
    actionTimeout: 0,
    baseURL,
    trace: 'on-first-retry',
  },
  reporter: [
    ['html', { outputFolder: '../../dist/apps/my-app-e2e/playwright-report' }],
    ['json', { outputFile: '../../dist/apps/my-app-e2e/playwright-report/test-results.json' }],
  ],
  fullyParallel: true,
  forbidOnly: !!process.env.CI,
  workers: process.env.CI ? 1 : undefined,
  projects: [
    {
      name: 'chromium',
      use: {
        ...devices['Desktop Chrome'],
      },
    },
    {
      name: 'firefox',
      use: {
        ...devices['Desktop Firefox'],
      },
    },
    {
      name: 'webkit',
      use: {
        ...devices['Desktop Safari'],
      },
    },
    /* Test against mobile viewports. */
    // {
    //   name: 'Mobile Chrome',
    //   use: {
    //     ...devices['Pixel 5'],
    //   },
    // },
    // {
    //   name: 'Mobile Safari',
    //   use: {
    //     ...devices['iPhone 12'],
    //   },
    // },
    /* Test against branded browsers. */
    // {
    //   name: 'Microsoft Edge',
    //   use: {
    //     channel: 'msedge',
    //   },
    // },
    // {
    //   name: 'Google Chrome',
    //   use: {
    //     channel: 'chrome',
    //   },
    // },
  ],
};

export default config;
"
`;

exports[`Playwright Project should set right path names in \`tsconfig.e2e.json\` 1`] = `
Object {
  "compilerOptions": Object {
    "allowJs": true,
    "outDir": "../../dist/out-tsc",
    "sourceMap": false,
    "types": Array [
      "node",
    ],
  },
  "extends": "./tsconfig.json",
  "include": Array [
    "src/**/*.ts",
    "src/**/*.js",
    "playwright.config.ts",
  ],
}
`;
